services/wallet/service.go

```diff
@@ -490,6 +490,12 @@ func (service *Service) LinkZebPayWallet(ctx context.Context, walletID uuid.UUID
 		return handlers.WrapError(errors.New(msg), msg, http.StatusBadRequest)
 	}
 
+	isValid, ok := claims["isValid"].(bool)
```

Comment:
Is this function mixing levels of abstraction i.e. decoding the token and also checking domain concepts?

Would it be make it more readable to decode the token then check the details i.e. is the user KYC and has valid details such as accountID and depositID. `isValid` could be quite ambiguous.

Reply:
Do we have any tests to show both a valid and failing request?

Reply:
added tests to show no kyc "isValid" should fail with bad request, and updated existing success case test to make sure isValid was there.